docs/user_manual.md

```diff
@@ -435,10 +435,11 @@ appears anywhere in the configuration. Semantically, it means that evaluation
 of that symbol will result in at most one return value (that is, the symbol is
 a *partial function*).
 
-The `functional` attribute indicates to the symbolic reasoning engine that a
-given symbol is a *total function*, that is it has *exactly* one return value
-for every possible input. Note that it does not make sense to have only the
-`functional` attribute without the `function` attribute.
+The `total` attribute can be added to a production with the `function`
+attribute to indicate to the symbolic reasoning engine that a given symbol is a
+*total function*, that is it has *exactly* one return value for every possible
+input. While the `total` attribute can in theory be added to something which
+is not a function, this behavior is not currently implemented.
```

Comment:
We should instead say that `total` means *at least* one return value for every possible input. With the additional restriction currently in place that it must be on a `function` production, we still get the same behavior (exactly one value for each input), and we can say as much. But it leaves us open to defining `total` symbols later that have _more_ than one value possible (think disjuncts on the RHS of rules).

Reply:
This also gives a nice symmetry with `function`, which means *at most* one return value for every possible input. And then `function, total` means *exactly* one return value per input.

Reply:
I guess what you have written doesn't exactly contradict this, but I think it could be made clearer.